docs/source/caching/normalized-cache.mdx

```diff
@@ -95,64 +128,77 @@ val apolloClient = ApolloClient.Builder()
+
+Whenever Apollo Android _writes_ data to the cache, those writes propagate down _all caches_ in the chain.
+
+## Setting a fetch policy
 
-## Getting data from the cache
+After you add a normalized cache to your `ApolloClient` initialization, Apollo Android automatically uses it by default for all queries. You can customize _how_ the cache is used for a particular query by setting a **fetch policy** for that query.
 
-Once your cache setup is complete, the cache will be used by default by all your queries. By default, queries will try to find a result in the cache first and go the network if it's not there. This is the `FetchPolicy.CacheFirst`. You can customize that behaviour with `fetchPolicy(FetchPolicy)`:
+The following snippet shows how to set all available fetch policies and their behavior:
 
 ```kotlin
 val response = apolloClient.query(query)
-                 // Don't use the cache at all
-                 .fetchPolicy(FetchPolicy.NetworkOnly)
 
-                 // Or use only the cache
-                 .fetchPolicy(FetchPolicy.CacheOnly)
+  // (Default) Check the cache, then only use the network if data isn't present
+  .fetchPolicy(FetchPolicy.CacheFirst)
 
-                 // Or use the cache only if the network fails
-                 .fetchPolicy(FetchPolicy.NetworkFirst)
+  // Check the cache and never use the network, even if data isn't present
+  .fetchPolicy(FetchPolicy.CacheOnly)
 
-                 // Or use the network only if the cache fails (default)
-                 .fetchPolicy(FetchPolicy.CacheFirst)
+  // Always use the network, then check the cache if network fails
+  .fetchPolicy(FetchPolicy.NetworkFirst)
 
-                 // In all cases, execute your query
-                 .execute()
+  // Always use the network and never check the cache, even if network fails
+  .fetchPolicy(FetchPolicy.NetworkOnly)
+
+  // Execute the query
+  .execute()
 ```
 
-Like for `normalizedCache(NormalizedCacheFactory)`, `fetchPolicy(FetchPolicy)` is an extension function on `ApolloClient.Builder()` so you will need `apollo-normalized-cache` in your classpath for this to work.
+As with `normalizedCache(NormalizedCacheFactory)`, `fetchPolicy(FetchPolicy)` is an extension function on `ApolloClient.Builder()`, so you need `apollo-normalized-cache` in your classpath for this to work.
 
-Because the normalized cache deduplicates all your data, it allows to react to cache changes. This is done with `watchers` that listen to cache changes. You can learn more about them [in this page](query-watchers). Using watchers requires deduplicating your objects by specifying their object ids.
+Because the normalized cache deduplicates data, it enables you to react to cache changes. You do thiswith `watchers` that listen to cache changes. [Learn more about query watchers.](./query-watchers/)
 
-## Specifying your object IDs
+## Specifying cache IDs
```

Comment:
Any rationale behind the use of `object id` vs `cache id` vs `cache key`? I tend to prefer `cache key` everywhere but maybe these terms have slightly different meanings?

Reply:
I think we should avoid "Object ID" to reduce overlap with "an object's `id`", which might or not be the ID for the cache.

Between "cache ID" and "cache key", I went with "cache ID" purely because we already use that in the React docs and it seemed like a reasonable tie-breaker. I do think that "cache key" is a very reasonable alternative and maybe we should flip it for _every_ client in the future